app.rb

```diff
@@ -0,0 +1,168 @@
+require './person'
+require './student'
+require './teacher'
+require './rental'
+require './classroom'
+
+class App
+  attr_accessor :books, :people
+
+  def initialize
+    @books = []
+    @people = []
+    @students = []
+    @teachers = []
+    @rentals = []
+  end
+
+  def run
+    user_input
+  end
+
+  def display_books
+    puts '----  List of All Books -----'
+    if @books.empty?
+      puts 'Book store is Empty !'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index + 1} - #{book.title} by #{book.author}"
+      end
+    end
+    puts '-----------------------------'
+  end
+
+  def list_of_people
+    puts '----  List of All People -----'
+    if @people.empty?
+      puts 'No People Registered !'
+    else
+      @people.each_with_index do |person, index|
+        if person.instance_of?(Student)
+          puts "#{index + 1} -  [Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+        else
+          puts "#{index + 1} - [Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+        end
+      end
+    end
+    puts '-----------------------------'
+  end
+
+  def create_person
+    print 'Which person do you want to create ? (1) for Teacher, (2) for Student : '
+    input = gets.chomp
+    case input.to_i
+    when 1
+      teacher_option
+    when 2
+      student_option
+    else
+      puts 'you entered invalid input! Please try again:'
+      create_person
+    end
+  end
+
+  def permission?(parent_permission)
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    case permission
+    when 'n', 'N'
+      !parent_permission
+    when 'y', 'Y'
+      parent_permission
+    else
+      permission?(parent_permission)
+    end
+  end
+
+  def create_student(age, classroom, name, parent_permission)
+    student = Student.new(age, classroom, name, parent_permission: parent_permission)
+    @people << student unless @people.include?(student)
+    @students << student unless @students.include?(student)
+  end
+
+  def create_teacher(specialization, age, name)
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher unless @people.include?(teacher)
+    @teachers << teacher unless @teachers.include?(teacher)
+  end
+
+  def student_option
+    print ' Enter student Name : '
+    name = gets.chomp
+    print ' Enter student Age : '
+    age = gets.chomp
+    print ' Enter student Classroom <number> : '
+    classroom = gets.chomp
+    parent_permission = true
+    permission?(parent_permission)
+    create_student(age.to_i, classroom.to_i, name, parent_permission)
+    puts "---------  New student Added!  ----------- \n
+    #{name} is #{age} years old in classroom #{classroom.to_i}"
+    puts '-------------------------------- '
+  end
+
+  def teacher_option
+    print ' Enter teacher Name : '
+    name = gets.chomp
+    print ' Enter teacher Age : '
+    age = gets.chomp
+    print ' Enter teacher specialization : '
```

Comment:
- [ ] Please Try and fix this mistake. I suggest that you follow the requirement well. 
- [ ] Your school library console app must behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) provided for you
- [ ] 
![Capture](https://user-images.githubusercontent.com/63972670/194700491-8b233fb9-3582-44d7-8929-b450e5306fbf.JPG)

- [ ] HINT: You can consider [opening this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point), It may help